astro-docs/src/content/docs/technologies/dotnet/Guides/migrate-from-nx-dotnet-core.mdoc

```diff
@@ -0,0 +1,448 @@
+{% /callout %}
+
+## Why Migrate?
+
+The official `@nx/dotnet` plugin represents a natural evolution, building on lessons learned from the community plugin while embracing native .NET workflows:
+
+### Better Performance
+
+Significant performance improvements through efficient MSBuild analysis:
+
+- **Efficient dependency detection** using MSBuild APIs instead of repeated CLI calls
+- **Smaller footprint** with minimal dependencies (2 packages vs 12+)
+
+### Simpler Architecture
+
+- **Embraces native .NET tooling**: Use `dotnet new`, `dotnet add`, etc. directly
+- **Command-based targets**: Transparent `dotnet` commands instead of custom executors
+- **Zero configuration**: Automatic project detection without custom config files
+- **Minimal learning curve**: If you know the .NET CLI, you're already an expert
+
+### Production-Ready Support
+
+- **Official Nx support**: Maintained by the Nx core team
+- **Always compatible**: Integrated into Nx releases, no version lag
+- **MSBuild accuracy**: Uses Microsoft's official APIs for more accurate project analysis
+- **Battle-tested**: Used in production Nx workspaces
+
+### Key Design Decisions
+
+The official plugin takes a streamlined approach:
+
+- **Project inference**: Automatic detection of .NET projects without configuration
+- **MSBuild integration**: Direct use of Microsoft's APIs for accurate dependency analysis
+- **Native CLI first**: Uses `dotnet` commands directly instead of custom wrappers
+- **Minimal abstraction**: Simpler implementation with fewer dependencies
+
+This results in a faster, more reliable, and easier to maintain solution.
+
+## Prerequisites
+
+Before starting the migration:
+
+- [ ] Upgrade to **.NET SDK 8.0 or later** (required)
+- [ ] Locate your current @nx-dotnet/core configuration (see below)
+- [ ] Review any custom generators and executors you're using
+- [ ] Ensure your projects build successfully with current setup
+
+### Locating Your Configuration
+
+The @nx-dotnet/core plugin configuration can be in one of two places:
+
+**Option 1: In `nx.json` (newer default)**
+
+```json
+{
+  "plugins": [
+    {
+      "plugin": "@nx-dotnet/core",
+      "options": {
+        "inferredTargets": { ... },
+        "solutionFile": "...",
+        // ... other options
+      }
+    }
+  ]
+}
+```
+
+**Option 2: In `.nx-dotnet.rc.json` (legacy)**
+
+```json
+{
+  "inferredTargets": { ... },
+  "solutionFile": "...",
+  // ... other options
+}
+```
+
+{% callout type="note" title="No Configuration?" %}
+If you don't have custom configuration in either location, that's fine! The plugin works with defaults, and migration will be even simpler.
+{% /callout %}
+
+## Migration Steps
+
+### 1. Install @nx/dotnet
+
+```shell {% frame="none" %}
+nx add @nx/dotnet
+```
+
+This automatically:
+
+- Adds `@nx/dotnet` to `package.json`
+- Registers the plugin in `nx.json`
+- Configures `production` named inputs
+- Updates `.gitignore`
```

Comment:
```suggestion
- Adds `@nx/dotnet` as a dependency
- Registers the plugin in `nx.json`
- Configures target inputs
- Updates `.gitignore`
```

is the encapsulated nx install popular with dotnet plugin? only reason I ask is bc calling out `package.json` and such might not match w/ mental model. but this is a small nit. feel free to ignore.

Reply:
Heh, I don't think the `.nx` install is very popular period. Nonetheless, its a good callout.